### PR TITLE
bug fixes

### PR DESCRIPTION
**Pull request by ChronoVortex**
- ~~fixed infinite loop due to unsigned int~~
- added missing return to hacking drone collision check
- drone derived classes were broken, replaced with SpaceDrone
- fixed SHIP prefix for HasEquipment

### CODE REVIEW

File: lua/LuaLibScript.h [hunk 1]
```diff
@@ -34,7 +34,7 @@ class LuaLibScript
         bool call_on_internal_chain_event_callbacks(InternalEvents::Identifiers, int nArg=0, int nRet=0);
 
         int call_on_render_event_pre_callbacks(RenderEvents::Identifiers, int nArg);
-        void call_on_render_event_post_callbacks(RenderEvents::Identifiers, unsigned int idx, int nArg);
```
Comment (Nasa62) [inline lua/LuaLibScript.h:37]: Why did you remove the `unsigned`? This index value is never negative, it's an array index, it must always be `>= 0`.

File: lua/LuaLibScript.cpp [hunk 2]
```diff
@@ -237,7 +237,7 @@ int LuaLibScript::call_on_render_event_pre_callbacks(RenderEvents::Identifiers i
 
     return idx;
 }
-void LuaLibScript::call_on_render_event_post_callbacks(RenderEvents::Identifiers id, unsigned int idx, int nArg)
```
Comment (Nasa62) [inline lua/LuaLibScript.cpp:240]: See other comment about `unsigned`

**Review by Nasa62 — CHANGES_REQUESTED** [review]
Changes about `unsigned`

File: lua/LuaLibScript.h (hunk 1 above)
Comment (Nasa62) [reply lua/LuaLibScript.h:37]: Or I guess, describe how this causes an infinite loop, because I don't understand what the heck you mean.

Comment (ChronoVortex) [reply lua/LuaLibScript.h:37]: The loop continues as long as idx >= 0, which is always true for an unsigned int.

Comment (Nasa62) [reply lua/LuaLibScript.h:37]: I'll need to look at my code, but this doesn't seem like the ideal way to fix it if it's a logic bug, the API should remain that it's an index number of an array which is bounded by 0.

Comment (ChronoVortex) [reply lua/LuaLibScript.h:37]: I was kind of afraid to touch the logic, I didn't completely get it. Shall I undo this change for this PR?

Comment (ChronoVortex) [reply lua/LuaLibScript.h]: I undid it

**Review by Nasa62 — APPROVED** [review]
